components/raftstore-v2/src/router/message.rs

```diff
@@ -95,49 +95,17 @@ impl StoreTick {
 }
 
 /// Command that can be handled by raftstore.
-pub struct RaftRequest {
+pub struct RaftRequest<C> {
```

Comment:
Why removing the RaftQuery and RaftCommand ?  And why not renaming it to RaftCommand (so that it's consistent with v1) ?

Reply:
Because the definition of `RaftQuery` and `RaftCommand` are the same except callback. So I use generic to simplify the code. Note `RaftQuery` and `RaftCommand` are still used in `PeerMsg`, so renaming `RaftRequest` to `RaftCommand` can be confusing. We are rewriting raftstore, so the consistency with v1 is not important, but writing more readable code is. As commented, `RaftQuery` is for those query that doesn't need to go through log commit process, `RaftCommand` is for those commands that need to be executed in the same order on all replicas. Both `RaftQuery` and `RaftCommand` are request sent from client, hence name the shared struct as `RaftRequest`.